osf/models/node.py

```diff
@@ -135,30 +136,26 @@ def can_view(self, user=None, private_link=None):
             qs |= self.filter(private_links__is_deleted=False, private_links__key=private_link)
 
         if user is not None and not isinstance(user, AnonymousUser):
-            if isinstance(user, OSFUser):
-                user = user.pk
-            if not isinstance(user, int):
-                raise TypeError('"user" must be either {} or {}. Got {!r}'.format(int, OSFUser, user))
-
-            sqs = Contributor.objects.filter(node=models.OuterRef('pk'), user__id=user, read=True)
-            qs |= self.annotate(can_view=models.Exists(sqs)).filter(can_view=True)
+            read_user_query = get_objects_for_user(user, 'read_node', self.filter(Q(contributor__user_id=user.id)))
+            qs |= read_user_query
             qs |= self.extra(where=["""
                 "osf_abstractnode".id in (
                     WITH RECURSIVE implicit_read AS (
-                        SELECT "osf_contributor"."node_id"
-                        FROM "osf_contributor"
-                        WHERE "osf_contributor"."user_id" = %s
-                        AND "osf_contributor"."admin" is TRUE
+                        SELECT osf_contributor.node_id
+                        FROM osf_contributor, auth_group, osf_osfuser_groups
+                        WHERE osf_contributor.user_id = %s
+                        AND auth_group.name = 'node_' || node_id || '_admin'
```

Comment:
This is an example of a query that works fine for just adding admin/write/read groups to nodes, but will need more thinking for OSF groups.  Here, I am checking group membership to see if you have admin permissions.  If you belong to a specific group "node_55_admin" for example, you must have admin permissions. 

If we are adding node permissions to OSF groups, then we couldn't just check this specific node group.